core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -161,6 +161,8 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
             TypeAnalyzer typeAnalyzer,
             DomainTranslator domainTranslator)
     {
+        verify(isAllowPushdownIntoConnectors(session), "Pushdown is disabled");
```

Comment:
This should just return Optional.empty(). 

Then, there's no special treatment needed for any of the callers of this method.

Reply:
It could -- yes, but not necessarily should. It's verify, since the caller is not expected to call this.
Note that this class already handles this via the `Rule.isEnabled()` method (so -- correctly), and only the "side  caller", the AddExchanges, needs the condition.

I am leaning towards your version though